pkg/config/validator.go

```diff
@@ -638,7 +639,12 @@ func (p *validator) buildHandler(builder *HandlerBuilderInfo, handler string) (c
 		}
 	}()
 
-	instance, err := (*builder.handlerBuilder).Build(builder.handlerCnfg.Params.(proto.Message), nil)
+	(*builder.handlerBuilder).SetAdapterConfig(builder.handlerCnfg.Params.(proto.Message))
```

Comment:
Having to deref a variable at every use is a strong signal it shouldn't be a pointer. Here the problem is that you have a pointer to an interface, which you almost never want in golang. `HandlerBuilderInfo.handlerBuilder` should be called `builder` and have type `adapter.HandlerBuilder` instead of `*adapter.HandlerBuilder`.

Reply:
You could even call the field `b` to avoid the `builder.builder` stutter: `builder.b.SetAdapterConfig(...); builder.b.Build(...)`

Reply:
Done